src/core/search/base.h

```diff
@@ -81,6 +81,13 @@ struct DocumentAccessor {
 // query functions. All results for all index types should be sorted.
 struct BaseIndex {
   virtual ~BaseIndex() = default;
+
+  /* Returns true if the field type in the document matches the index and, therefore, the document
+  can be added.
+  TODO: Return the data needed for the Add() function to avoid retrieving the same data twice.*/
+  virtual bool Matches(DocId id, DocumentAccessor* doc, std::string_view field) = 0;
+
+  // Before adding the document make sure that Matches() returns true
   virtual void Add(DocId id, DocumentAccessor* doc, std::string_view field) = 0;
```

Comment:
I understand that we could implement it as `bool Add()`. I did it this way because, by the next release, we might be able to implement `std::optional<Data> Matches()`, with `Add` taking this data as an argument. This way, we won’t need to do a rollback, and we’ll fetch the data only once. Moreover, we already have a cache in `JsonAccessor`.